Modules/_csv.c

```diff
@@ -1606,8 +1606,9 @@ _csv_unregister_dialect_impl(PyObject *module, PyObject *name)
 /*[clinic end generated code: output=0813ebca6c058df4 input=6b5c1557bf60c7e7]*/
 {
     _csvstate *module_state = get_csv_state(module);
-    if (PyDict_DelItem(module_state->dialects, name) < 0) {
-        if (PyErr_ExceptionMatches(PyExc_KeyError)) {
+    int rc = PyDict_Pop(module_state->dialects, name, NULL);
+    if (rc <= 0) {
+        if (rc == 0) {
```

Comment:
I would prefer to avoid nested if here. Just write "if (rc < 0) ..." and "if (rc == 0) ...".